### DISCUSSION TITLE
Feasycom FSC-BP309 - USB Super Long Range USB-Bluetooth Transceiver

### DISCUSSION THREAD

**Discussion by Arman-GeDon** [custom-components/ble_monitor#305] [Q&A]
Does anyone use this device: https://www.feasycom.com/product-usb-super-long-range-usb-bluetooth-transceiver.html

It's seen as serial device connected to virtual COM port and not as normal USB bluetooth dongle so I don't know how to use it with ble monitor

- name: ttyUSB0
    sysfs: '/sys/devices/pci0000:00/0000:00:06.0/usb2/2-2/2-2:1.0/ttyUSB0/tty/ttyUSB0'
    dev_path: /dev/ttyUSB0
    subsystem: tty
    by_id: >-
      /dev/serial/by-id/usb-Silicon_Labs_CP2104_USB_to_UART_Bridge_Controller_019C4D1F-if00-port0
    attributes:
      DEVLINKS: >-
        /dev/serial/by-id/usb-Silicon_Labs_CP2104_USB_to_UART_Bridge_Controller_019C4D1F-if00-port0
        /dev/serial/by-path/pci-0000:00:06.0-usb-0:2:1.0-port0
      DEVNAME: /dev/ttyUSB0
      DEVPATH: '/devices/pci0000:00/0000:00:06.0/usb2/2-2/2-2:1.0/ttyUSB0/tty/ttyUSB0'
      ID_BUS: usb
      ID_MODEL: CP2104_USB_to_UART_Bridge_Controller
      ID_MODEL_ENC: CP2104\x20USB\x20to\x20UART\x20Bridge\x20Controller
      ID_MODEL_ID: ea60
      ID_PATH: 'pci-0000:00:06.0-usb-0:2:1.0'
      ID_PATH_TAG: pci-0000_00_06_0-usb-0_2_1_0
      ID_REVISION: '0100'
      ID_SERIAL: Silicon_Labs_CP2104_USB_to_UART_Bridge_Controller_019C4D1F
      ID_SERIAL_SHORT: 019C4D1F
      ID_TYPE: generic
      ID_USB_DRIVER: cp210x
      ID_USB_INTERFACES: ':ff0000:'
      ID_USB_INTERFACE_NUM: '00'
      ID_VENDOR: Silicon_Labs
      ID_VENDOR_ENC: Silicon\x20Labs
      ID_VENDOR_ID: 10c4
      MAJOR: '188'
      MINOR: '0'
      SUBSYSTEM: tty
      TAGS: ':systemd:'
      USEC_INITIALIZED: '135402253020'

**Comment by Chao-chao00** [comment]
Hi i just saw this topic. I have the same Feasycom FSC-BP309 and it seems to be designed not to work as a traditional USB Dongle for computers but more like plugging it into a device that use serial communication to give it wireless functionality (like a barcode scanner).

The computer only sees it as a serial device so you can send AT commands (for example AT+SCAN=1) to the FSC-BP309 to perform actions. It is based on the FSC-BT909 chipset and you have a programming user guide that lists commands that you can send to it, i use the Feasycom Serial Port Windows app:
https://www.feasycom.com/programming-user-guide/fsc-bt90x.pdf
https://www.feasycom.net/news/bluetooth-module-windows-app-feasycom-serial-25381850.html
I tried some commands but i still don't understand why i can scan BR/EDR normal Bluetooth devices but i cannot scan BLE devices (returns nothing)

I have a specific use case for the FSC-BP309 with Home Assistant and NodeRed. Since a recent Home Assistant version, the official Bluetooth Tracker integration doesn't work anymore (the BLE one still works) so it is impossible to scan for normal bluetooth devices with the Home Assistant bluetooth.
https://community.home-assistant.io/t/bluetooth-tracker-not-working-at-all/587993

I have an audio preamp that has 3 inputs (RCA, XLR and Bluetooth) and 1 output, it has an IR remote and you can select the input but there is only one button on the remote that cycles through the 3 inputs. I don't use the Bluetooth on this preamp and when you select the Bluetooth input it broadcast its address and name "BT HIFI AUDIO" and when you select another input the Bluetooth radio is turned off.

So to automate this device Input selection i use NodeRed to send the AT command to the FSC-BP309 (plugged into my Home Assistant machine) to scan for Bluetooth devices (AT+SCAN=1), NodeRed then receives the answer via serial, after 10s if the "BT HIFI AUDIO" is not found then Home Assistant uses my Tasmota IR remote to send the IR code to cycle the input. Repeat this scanning/changing inputs process until you find the "BT HIFI AUDIO" device and then you know that you are currently in the Bluetooth input mode and you can easily change to other modes because you know the cycle order.

Note : I asked in a email about why the BLE Scan doesn't work and it got this answer : "No BLE devices are detected during scanning because the firmware only supports the module as a GATT server, not as a GATT client."
Indeed trying to change to GATT client using the AT+PROFILE command doesn't work because of the firmware restrictions.
